lib/rubygems/gemcutter_utilities.rb

```diff
@@ -113,12 +123,31 @@ def sign_in sign_in_host = nil
       request.basic_auth email, password
     end
 
+    if need_mfa? response
+      check_mfa
```

Comment:
`check_mfa` and `need_mfa?` always seem to go together. Can we please call `check_mfa` from `need_mfa?` instead?